dbms/src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -1612,9 +1623,36 @@ void InterpreterSelectQuery::executeFetchColumns(
             }
         }
 
+        /// Copy-paste from prev if.
+        if (pipes.empty() && use_pipes)
+        {
+            Pipe pipe(std::make_shared<NullSource>(storage->getSampleBlockForColumns(required_columns)));
+
+            if (query_info.prewhere_info)
```

Comment:
Why sometimes we use `query_info.prewhere_info` and sometimes `prewhere_info`? Looks like it's the same pointer.

Reply:
Yes. It's the same. We do `query_info.prewhere_info = prewhere_info;`